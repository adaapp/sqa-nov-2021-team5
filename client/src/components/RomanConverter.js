import React, {useState} from 'react';
import Axios from 'axios';
import '../App.css';

function RomanConverter(){
    const [numbers, setNumbers] = useState({romanNumber: "", normalNumber: "", action: ""})

    
    const romanDecoder = () => {
        Axios.post("http://localhost:5000/roman-decoder", {romanNumber: numbers.romanNumber}).then((response => {
            setNumbers({romanNumbmer: numbers.romanNumber, normalNumber: response.data, action: "roman-decode"})
        }))
    }

    const romanEncoder = () => {
        Axios.post("http://localhost:5000/roman-encoder", {normalNumber: numbers.normalNumber}).then((response => {
            setNumbers({normalNumber: numbers.normalNumber, romanNumber: response.data, action: "roman-encode"})
        }))
    }

    const convert = () => {
        if(numbers.action == "roman-decode"){
            romanDecoder()
        }
        if(numbers.action == "roman-encode"){
            romanEncoder()
        }
    }

    const clear = () => {
        setNumbers({romanNumber: "", normalNumber: "", action: ""})
    }


    return  <div id="root">
               
                <div>
                    <h1>Roman numerals converter</h1>
                </div>
                <div>
                 
                        {(!numbers.romanNumber && !numbers.normalNumber)
                        ?   <div>   
                                <input type="text" placeholder="Normal Number" value={numbers.normalNumber} onChange={(event) => {setNumbers({normalNumber: event.target.value, action: "roman-encode"})}}/> 
                                <input type="text" placeholder="Roman Number" value={numbers.romanNumber} onChange={(event) => {setNumbers({romanNumber: event.target.value, action: "roman-decode"})}}/>
                            </div>
                        
                        : (numbers.action === "roman-encode" && numbers.normalNumber)

                            ?   <div>   
                                    <input type="text" placeholder="Normal Number" onChange={(event) => {setNumbers({normalNumber: event.target.value, action: "roman-encode"})}}/> 
                                    <input type="text" placeholder="Roman Number" disabled={true} value={numbers.romanNumber}/>
                                </div>


                            :   <div>
                                    <input type="text" placeholder="Normal Number" disabled={true} value={numbers.normalNumber}/>
                                    <input type="text" placeholder="Roman Number" onChange={(event) => {setNumbers({romanNumber: event.target.value, action: "roman-decode"})}}/>
                                </div>
                    }
                  
                </div>
                <div>
                    <button onClick={convert}>Convert</button>
                    <button onClick={clear}>Clear</button>
                </div>
            </div>
}

export default RomanConverter;