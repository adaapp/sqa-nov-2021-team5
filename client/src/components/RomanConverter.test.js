import { shallow } from "enzyme"
import toJSON from "enzyme-to-json"
import React from "react"

import RomanConverter from "./RomanConverter"
import Button from "./RomanConverter"
import Input from "./RomanConverter"

describe("Smoke test", () => {
    it("1 + 1 equals 2", () => {
        expect(1+1).toEqual(2)
    })
})


describe("Rendering Test", () => {
    test("Test to check, if component renders correctly", () => {
    const wrapper = shallow(<RomanConverter/>)

    expect(toJSON(wrapper)).toMatchSnapshot()
})
})

describe('Button', () => {
    
    test('should be defined', () => {
      expect(Button).toBeDefined();
    });
    
    test('test to check, if button renders correctly', () => {
      const wrapper = shallow(
        <Button name='button test' />
      );
      expect(wrapper).toMatchSnapshot();
    });
   });

   describe('Input', () => {
       test('should be defined', () => {
           expect(Input).toBeDefined()
       })

       test('test to check, if input field renders correctly', () => {
        const wrapper = shallow(
          <Input name='input test' />
        );
        expect(wrapper).toMatchSnapshot();
      });
   })


