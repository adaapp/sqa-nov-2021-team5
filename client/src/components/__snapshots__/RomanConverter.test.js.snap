// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test to check, if button renders correctly 1`] = `
<div
  id="root"
>
  <div>
    <div>
      <input
        className="Input-roman-number"
        onChange={[Function]}
        placeholder="Roman Number"
        type="text"
        value=""
      />
      <input
        className="Input-normal-number"
        onChange={[Function]}
        placeholder="Normal Number"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Convert
    </button>
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
</div>
`;

exports[`Input test to check, if input field renders correctly 1`] = `
<div
  id="root"
>
  <div>
    <div>
      <input
        className="Input-roman-number"
        onChange={[Function]}
        placeholder="Roman Number"
        type="text"
        value=""
      />
      <input
        className="Input-normal-number"
        onChange={[Function]}
        placeholder="Normal Number"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Convert
    </button>
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
</div>
`;

exports[`Rendering Test Test to check, if component renders correctly 1`] = `
<div
  id="root"
>
  <div>
    <div>
      <input
        className="Input-roman-number"
        onChange={[Function]}
        placeholder="Roman Number"
        type="text"
        value=""
      />
      <input
        className="Input-normal-number"
        onChange={[Function]}
        placeholder="Normal Number"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Convert
    </button>
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
</div>
`;
